### PR TITLE
fix: DO NOT MERGE: stkOSMO migration

### PR DESCRIPTION
**Pull request by kruspy**

### CODE REVIEW

**Comment by kruspy** [comment]
The upgrade test keeps failing. I haven't seen this before and it looks like it is not related with the test itself.

@xlab

**Comment by xlab** [comment]
> The upgrade test keeps failing. I haven't seen this before and it looks like it is not related with the test itself.
> 
> @xlab

Image `persistenceone/persistencecore:v10.0.0` has been created on `arm64` machine while it was expected to be done on `x86_64` machine, I will reupload.

**Comment by puneet2019** [comment]
I remember commenting this, but don't know where it went.

> LGTM, can we make sure while tagging we tag it with `testnet` in the title, it might be confusing for validators since this will never go on mainnet. 
And @kruspy are you sure you have tested this, because SDK docs ask to not [write while iterating](https://github.com/persistenceOne/cosmos-sdk/blob/v47-lsm/store/types/store.go#L232-L233)

**Comment by xlab** [comment]
https://github.com/persistenceOne/persistenceCore/pull/253#issuecomment-1781147134

Makes sense, just wanted to keep it as closest to semver naming as possible.
`v10.1.0-rc1` or `v10.1.0-testnet` will do.

**Review by xlab — APPROVED** [review]
LGTM

File: app/upgrades/v10.1.0-testnet/upgrades.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		mStore := prefix.NewStore(ctx.KVStore(args.Keepers.GetKey(liquidstakeibctypes.StoreKey)), liquidstakeibctypes.DepositKey)
+		iterator := sdk.KVStorePrefixIterator(mStore, nil)
+		defer iterator.Close()
+
+		for ; iterator.Valid(); iterator.Next() {
+			deposit := liquidstakeibctypes.Deposit{}
+			args.Codec.MustUnmarshal(iterator.Value(), &deposit)
+
+			if deposit.ChainId == HostChainID {
+				mStore.Delete(liquidstakeibctypes.GetDepositStoreKey(deposit.ChainId, deposit.Epoch))
```
Comment (xlab) [inline app/upgrades/v10.1.0-testnet/upgrades.go:60]: A valid concern raised by @puneet2019  – this write must be outside the iterator. Collect records into an array and delete them after iteration is done.

File: app/upgrades/v10.1.0-testnet/upgrades.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		mStore = prefix.NewStore(ctx.KVStore(args.Keepers.GetKey(liquidstakeibctypes.StoreKey)), liquidstakeibctypes.LSMDepositKey)
+		iterator = sdk.KVStorePrefixIterator(mStore, nil)
+		defer iterator.Close()
+
+		for ; iterator.Valid(); iterator.Next() {
+			deposit := liquidstakeibctypes.LSMDeposit{}
+			args.Codec.MustUnmarshal(iterator.Value(), &deposit)
+
+			if deposit.ChainId == HostChainID {
+				mStore.Delete(liquidstakeibctypes.GetLSMDepositStoreKey(deposit.ChainId, deposit.DelegatorAddress, deposit.Denom))
```
Comment (xlab) [inline app/upgrades/v10.1.0-testnet/upgrades.go:74]: same here and other places

**Review by xlab — CHANGES_REQUESTED** [review]
Ah, iterator still writes

**Comment by puneet2019** [comment]
@kruspy can we close this PR?

**Pull request closed without merging**